### ISSUE TITLE
Funding Credit on Collections pages

### ISSUE THREAD

**Issue by apinkney0696** [sciencehistory/scihist_digicoll#1526]
Related to ticket #1358 

Create collection-level metadata fields to credit an external funder: 

1. a field for the funder name
2. a field for a link to the funder website if applicable
3. and a field to upload an image of the funding organization's logo.

The collections page should display the text "Digitization and cataloging of this collection made possible through the generosity of [name of donor or funder]" (hyperlink the name of the funding agent if possible, otherwise, add the link to the page separately), and the image of the funder's logo.

As far as the page layout goes, see the attached image for an idea. I'm open to different ideas and playing around to make things fit. The mobile/compressed view would simply stack all of the elements.
![IMG_1962](https://user-images.githubusercontent.com/78036563/148087968-9f6b847a-2361-4c99-a3e9-26f53e268d93.jpeg)

**Comment by jrochkind** [comment]
I think we previously discussed that as an implementation efficciency, we might not really allow upload of the image, but just have a fixed set of images in the sourcecode, that can only be changed by developers (me and Eddie). Since we don't expect very many of these to be added very often. 

We also briefly considered having a "funder" object in the database as a separate normalized entity, so you'd edit/add/remove "funders" in a separate place, then just choose from the list of "funders"  in the collection. But I think we decided _not_ to do that, and that it was unlikely funders would be re-used much, and  okay to just repeat data when they were, no need to complicate the data model for it

**Comment by jrochkind** [comment]
@apinkney0696 is "Digitization and cataloging of this collection made possible by the generosity of" fixed text, that would automatically appear for any collection using this feature, the metadata is just "CLIR", then that fixed text is added in front of it on display?

**Comment by apinkney0696** [comment]
> I think we previously discussed that as an implementation efficciency, we might not really allow upload of the image, but just have a fixed set of images in the sourcecode, that can only be changed by developers (me and Eddie). Since we don't expect very many of these to be added very often.

That's right! I forgot about that. So that part of the ticket request should be adjusted to developers fix the logo images to the pages. I'll have to compile a list of the collections that are going to need this work and their corresponding logo images for you.

> We also briefly considered having a "funder" object in the database as a separate normalized entity, so you'd edit/add/remove "funders" in a separate place, then just choose from the list of "funders" in the collection. But I think we decided not to do that, and that it was unlikely funders would be re-used much, and okay to just repeat data when they were, no need to complicate the data model for it

Yes, I do remember talking about this. Because there is likely just going to be one funder per collection, I don't think it will be necessary to add a drop-down menu.

> is "Digitization and cataloging of this collection made possible by the generosity of" fixed text, that would automatically appear for any collection using this feature, the metadata is just "CLIR", then that fixed text is added in front of it on display?

Ideally yes, that text would be fixed in the display and the funding organization's name would be the only thing entered into the metadata.

**Comment by apinkney0696** [comment]
Some more info from me. Below is the information for the collections which have a funder. The logos can be found [here](https://files.sciencehistory.org/app/index.do#storage/files/1/Shared/P/Othmer%20Library/Digital%20Collections%20-%20Internal%20Access/Website%20Images/Funder%20Logos). I was able to download most from online, but consequently they vary between .png and .jpg formats. Let me know if you need me to do any format conversions.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Annabel/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Annabel/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">



Collections   with a Funder |   |  
-- | -- | --
Collection | Funding Organization | URL
Papers of Georg and Max Bredig | Council on Library and Information Resources | https://www.clir.org/
Beckman Historical Collection | Arnold & Mabel Beckman Foundation | https://www.beckman-foundation.org/
Chemical History of Electronics Oral   History Project | Gordon and Betty Moore Foundation | https://www.moore.org/
Imagining Philadelphia's Energy Futures   Oral History Project | Climate & Urban Systems Partnership | http://www.cuspproject.org/
Life Sciences Foundation Oral History   Project | Life Sciences Foundation | https://www.sciencehistory.org/lsf
Mass Spectrometry Oral History Project | American Society for Mass Spectrometry | https://www.asms.org/
Nanotechnology Oral History Project | Center for Nanotechnology in Society | http://cns.asu.edu/
Papers of James Curtis Booth | Newman Numismatic Portal at Washington University in St. Louis | https://nnp.wustl.edu/
Pew Biomedical Scholars Oral History   Project | Pew Charitable Trusts | https://www.pewtrusts.org/en/
REACH Ambler Oral History Project | National Institutes of Health (U.S.) | https://www.nih.gov/
Scientific and Technical Information   Systems Oral History Project | Eugene Garfield Foundation | n/a
Immigration and Innovation Oral History   Project | National   Historical Publications & Records Commission | https://www.archives.gov/nhprc



</body>

</html>

**Comment by jrochkind** [comment]
We discussed in Slack that we're going to require SQUARE icons, so we can lay it out so lovely as in mock-up above. 

* JPG or PNG is fine. 
* No transparent backgrounds in PNGs though please -- or, I mean, there can be transparent bg around the outside of a shape, that's fine. But one of those examples was just WORDS on a transaparent background, which isn't going to work well if the background color is too close in size to the words! Transparent bg only that will work with arbitrary bg color, if you want to think about it. 
* Not sure I am sure about exactly how many pixels the image should be, but I'm going to say 224x224px if that works?  A little bit bigger is fine, but not a lot smaller, and definitely no smaller than 112px x 112px.

**Comment by apinkney0696** [comment]
Some other updates:

For collections with "Oral History" set as their collecting area, the automatic text should read 
> This oral history project was made possible by the generosity of

Collections with collecting areas "Museum," "Library," or "Archives" should have the automatic text
> Digitization and cataloging of this collection was made possible by the generosity of

Only other feedback is to add a few px of white space on either side of the square logo space
![image](https://user-images.githubusercontent.com/78036563/149397661-9cb23187-89e0-4d14-8f53-dc4c845893b8.png)

**Comment by jrochkind** [comment]
Aha! @apinkney0696 thanks for including the screenshot, because  that is NOT what I see on my browser or what I intended!

Here it is for me as I intended (which still maybe still could use a bit more whitespace!)

![Screen Shot 2022-01-13 at 3 15 22 PM](https://user-images.githubusercontent.com/149304/149402671-f7e54450-d72d-49b4-8503-230132d9803d.png)


It's looking not as intended on your browser. I thought I tested on all browsers, but clearly not! Can you tell me your OS, browser, and versiion of that browser you are using?

**Comment by jrochkind** [comment]
New version of icons at: https://files.sciencehistory.org/app/index.do#storage/files/1/Shared/P/Othmer%20Library/Digital[…]0Internal%20Access/Website%20Images/Funder%20Logos

**Comment by jrochkind** [comment]
@apinkney0696 the new images look good thanks!

Most of them are reasonably sized (under 100KB)-- although they probably could be even smaller if we spent time on looking at JPG compression level and such, I think they're good enough. 

But the Gordon and Betty Moore one is for some reason 629KB...  converting it to a PNG instead of a JPG brings it down to 61KB, and it still looks indistinguishable to me, so I'm going to go ahead and do that.... 

I might actually run all of them through an optimizer to reduce file size while hopefully not effecting image quality, if you don't mind.

**Comment by jrochkind** [comment]
Thanks! It may be the issue, but I want it to work on someone's browser which is a bit out of date too, I'll investigate more!

**Comment by jrochkind** [comment]
OK, I succesfully reproduced problem on Chrome 80. (using browserstack I could run as MacOS Chrome 80, hooray). 

And saw another problem too with this stuff. 

I'll get it working right on Chrome 80, cause, that's only a year old, and if you had it others might.
